sys/compat/freebsd64/freebsd64_uipc.c

```diff
@@ -287,14 +301,23 @@ freebsd64_copyin_control(struct mbuf **mp, char * __capability buf,
 		newlen += CMSG_SPACE(datalen);
 	}
 
-	if (newlen > MCLBYTES)
-		return (EINVAL);
-
-	m = m_get2(newlen, M_WAITOK, MT_CONTROL, 0);
```

Comment:
Even shorter though here would be to just use `m_get3` which will allocate larger mbufs if needed from the jumbo frame zones.  That's probably the simplest diff to do doesn't require a custom EXT_* constant, etc.  (I didn't think of just using a jumbo frame mbuf before.)

Reply:
I went to look at the manpage entry for `m_get3`, found it missing, and read the `m_get2` description. I think the fix may be even simpler. Just change the guard be `newline > MJUMPAGESIZE` to match m_get2's limit.

Reply:
Oh, huh.  For some reason I had thought `m_get2` would only allocate regular clusters and not page-sized clusters.  I agree that just updating the limit would be sufficient.